finality-aleph/src/network/component.rs

```diff
@@ -10,12 +11,73 @@ pub trait Sender<D: Data>: Sync + Send + Clone {
     fn send(&self, data: D, recipient: Recipient) -> Result<(), SendError>;
 }
 
+#[derive(Clone)]
+pub struct MapSender<D, S> {
+    sender: S,
+    _phantom: PhantomData<D>,
+}
+
+pub trait SenderMap<From: Data>: Sender<From> {
```

Comment:
nit: rename From to something else as From is already a trait in rust and is a bit misleading